### PR TITLE
Adds flexible OSD Sizes in the expand and install flow

### PR DESCRIPTION
**Pull request by afreen23**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1783231 
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1792778
Screenshot from 2020-01-22 18-33-40
Screenshot from 2020-01-22 18-33-55

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss [hunk 1]
```diff
@@ -5,33 +5,31 @@
...
 .add-capacity-modal--padding {
   .co-storage-class-dropdown {
     .dropdown {
       width: 340px;
-        button {
-          width: 100%;
-        }
+      button {
+        width: 100%;
+      }
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss]: Fox this

**Comment by afreen23** [comment]
This needs some more discussion. 
Holding till that.

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 2]
```diff
@@ -3,44 +3,69 @@ import * as _ from 'lodash';
...
+    ocsConfig,
+    'spec.storageDeviceSets[0].dataPVCTemplate.spec.resources.requests.storage',
+  );
+  const osdSizeWithoutUnit = OSD_CAPACITY_SIZES[osdSizeWithUnit].size;
+
+  const [requestSizeValue, setRequestSizeValue] = React.useState(osdSizeWithoutUnit);
+  const [response, loadError, loading] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint.QUERY,
+    namespace: CEPH_STORAGE_NAMESPACE,
+    query: 'ceph_cluster_total_used_bytes',
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: import this from the query file

File: ~1 [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# .       message (or the oneline, if no original merge commit was
+# .       specified). Use -c <commit> to reword the commit message.
+#
+# These lines can be re-ordered; they are executed from top to bottom.
+#
+# If you remove a line here THAT COMMIT WILL BE LOST.
+#
+# However, if you remove everything, the rebase will be aborted.
+#
+# Note that empty commits are commented out
```
Comment (afreen23) [inline ~1]: junk, will remove

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss [hunk 4]
```diff
@@ -1,37 +1,31 @@
...
-  max-width: 250px !important;
+.ceph-add-capacity__current-capacity--loading {
+  width: 7.9rem;
+  margin-top: 0.2rem;
 }
 
-.add-capacity-modal__input .dropdown {
-  margin-left: var(--pf-global--spacer--md);
+.ceph-add-capacity__current-capacity--text {
+  margin-right: 0.5rem;
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss]: sm din't work?

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss [hunk 5]
```diff
@@ -1,37 +1,31 @@
...
-  color: $pf-color-red-100;  
+.ceph-add-capacity__form {
+  display: flex;
+  align-items: center;
 }
 
-.add-capacity-modal__help-text {
-  margin-top: -15px;
+.ceph-add-capacity__input {
+  max-width: 250px !important;
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss]: `!important` should not be required? Could you please explain where the style clash is happening?

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 6]
```diff
@@ -55,76 +80,56 @@ export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps)
...
-              Please enter an even number
-            </p>
+            {requestSizeValue && (
+              <div className="ceph-add-capacity__input--info-text">
+                x 3 replicas ={' '}
+                <strong>
+                  {requestSizeValue % 1 ? (requestSizeValue * 3).toFixed(2) : requestSizeValue * 3}{' '}
+                  TiB
+                </strong>{' '}
+                {/* float precision issue hence rounding to 2 decimal places */}
```
[the remark this answers is not in the document]
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: What ?

File: ~1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (afreen23) [reply ~1]: already mentioned

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss (hunk 4 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss]: yeah,

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 7]
```diff
@@ -55,76 +80,56 @@ export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps)
...
-                'add-capacity-modal__help-text-error': buttonDisabled,
-              })}
-            >
-              Please enter an even number
-            </p>
+            {requestSizeValue && (
+              <div className="ceph-add-capacity__input--info-text">
+                x 3 replicas ={' '}
+                <strong>
+                  {requestSizeValue % 1 ? (requestSizeValue * 3).toFixed(2) : requestSizeValue * 3}{' '}
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: can you put this a function

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss (hunk 5 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss]: Yes, this was present before hand, hence did not touch.
Will look into it, and do whats needful.

File: frontend/packages/ceph-storage-plugin/src/utils/osd-size-dropdown.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    />
+  );
+};
+
+type DropdownOptions = { [key: string]: React.ReactNode };
+
+type OSDSizeDropdownProps = {
+  className: string;
+  selectedKey: string;
+  onChange: (value: string | ((prevVar: string) => string)) => void;
```
[the remark this answers is not in the document]
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/utils/osd-size-dropdown.tsx]: To set the selected key in dropdown

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: the comment.

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx (hunk 2 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: I need to then import three objects for that.

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss (hunk 5 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/_add-capacity-modal.scss]: yes, some generic css is having `max-width: 100px` for all input sizes of number type.
Hence `!important` is used to override that behavior

File: frontend/packages/ceph-storage-plugin/src/utils/osd-size-dropdown.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/utils/osd-size-dropdown.tsx]: please extract the  `prevVar` based function to a diff type

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx (hunk 2 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: Done

**Comment by afreen23** [comment]
> Remove all the _.get and use Optional Chaining ts 3.7

That will be a pain in eye! Will do

File: frontend/packages/ceph-storage-plugin/src/utils/osd-size-dropdown.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/utils/osd-size-dropdown.tsx]: This gets the fucntion returned by react hook, hence I wrote this way.
In recent versions, the types are modernized, I have made a change for this

**Pull request merged**